src/logic/embedding.lean

```diff
@@ -34,6 +34,17 @@ end function
 protected def equiv.to_embedding {α : Sort u} {β : Sort v} (f : α ≃ β) : α ↪ β :=
 ⟨f, f.injective⟩
 
+/-- Given an equivalence to a subtype, produce an embedding to the elements of the corresponding
+set. -/
+@[simps]
+def equiv.as_embedding {α β : Sort*} {p : β → Prop} (e : α ≃ subtype p) : α ↪ β :=
+⟨coe ∘ e, subtype.coe_injective.comp e.injective⟩
```

Comment:
Github marked https://github.com/leanprover-community/mathlib/pull/7037#discussion_r607051336 as outdated, which discusses this name.

Reply:
Unrelated to the name conversation, we have some API to construct this:
```suggestion
/-- Given an equivalence to a subtype, produce an embedding to the elements of the corresponding
set. -/
@[simps]
def equiv.as_embedding {α β : Sort*} {p : β → Prop} (e : α ≃ subtype p) : α ↪ β :=
e.to_embedding.trans $ function.embedding.subtype p
```
which may or may not break the proof below.